include/envoy/thread_local/thread_local.h

```diff
@@ -46,6 +46,14 @@ class Slot {
    */
   virtual void runOnAllThreads(Event::PostCb cb) PURE;
 
+  /**
+   * Run a callback on all registered threads with a barrier. There is no guarantee that the
+   * callback be fired during shutdown.
```

Comment:
Again really just to clarify semantics, would it be correct to rephrase this:

A shutdown initiated during the running of the PostCBs may prevent main_callback from being called.

WDYT?

Reply:
Sure